test/discr/discr_mappings_test.py

```diff
@@ -0,0 +1,392 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import pytest
+import numpy as np
+
+# Internal
+import odl
+from odl.discr.grid import sparse_meshgrid
+from odl.discr.discr_mappings import (
+    GridCollocation, NearestInterpolation, LinearInterpolation,
+    PerAxisInterpolation)
+from odl.util.testutils import all_almost_equal, all_equal, almost_equal
+
+
+def test_nearest_interpolation_1d_complex():
+    intv = odl.Interval(0, 1)
+    grid = odl.uniform_sampling(intv, 5, as_midp=True)
+    # Coordinate vectors are:
+    # [0.1, 0.3, 0.5, 0.7, 0.9]
+
+    space = odl.FunctionSpace(intv, field=odl.ComplexNumbers())
+    dspace = odl.Fn(grid.ntotal, dtype='complex128')
+    interp_op = NearestInterpolation(space, grid, dspace)
+    values = np.arange(5) + 1j * np.arange(1, 6)
+    function = interp_op(values)
+
+    # Evaluate at single point
+    val = function(0.35)  # closest to index 1 -> 1 + 2j
+    assert val == 1.0 + 2.0j
+    # Input array, with and without output array
+    pts = np.array([0.4, 0.0, 0.65, 0.95])
+    true_arr = [1 + 2j, 0 + 1j, 3 + 4j, 4 + 5j]
+    assert all_equal(function(pts), true_arr)
+    # Should also work with a (1, N) array
+    pts = pts[None, :]
+    assert all_equal(function(pts), true_arr)
+    out = np.empty(4, dtype='complex128')
+    function(pts, out=out)
+    assert all_equal(out, true_arr)
+    # Input meshgrid, with and without output array
+    # Same as array for 1d
+    mg = sparse_meshgrid([0.4, 0.0, 0.65, 0.95])
+    true_mg = [1 + 2j, 0 + 1j, 3 + 4j, 4 + 5j]
+    assert all_equal(function(mg), true_mg)
+    function(mg, out=out)
+    assert all_equal(out, true_mg)
+
+
+def test_nearest_interpolation_1d_variants():
+    intv = odl.Interval(0, 1)
+    grid = odl.uniform_sampling(intv, 5, as_midp=True)
+    # Coordinate vectors are:
+    # [0.1, 0.3, 0.5, 0.7, 0.9]
+
+    space = odl.FunctionSpace(intv)
+    dspace = odl.Fn(grid.ntotal, dtype='float64')
+
+    # 'left' variant
+    interp_op = NearestInterpolation(space, grid, dspace, variant='left')
+    values = np.arange(5, dtype='float64')
+    function = interp_op(values)
```

Comment:
The `dtype` is only needed for `arange` since it produces an `int` array by default. The operator would convert, but maybe the test code relying on the array would have undesired results. I'll just change to the simple list.